src/core/context/ThemeContext.tsx

```diff
@@ -1,60 +1,60 @@
-import { CssBaseline, ThemeProvider as MuiThemeProvider, useMediaQuery } from '@mui/material';
-import lightTheme from '../../../styles/theme/light';
+import { CssBaseline, ThemeProvider as MuiThemeProvider } from '@mui/material';
+import { createContext, ReactNode, useContext, useMemo, useEffect } from 'react';
 import darkTheme from '../../../styles/theme/dark';
-
-import { itemsWebSiteLinks } from '../../stories/components/header/select-link-website/menu-items';
-import Header from '../../stories/components/header/header';
+import lightTheme from '../../../styles/theme/light';
+import styled from '@emotion/styled';
 import Footer from '../../stories/components/footer/footer';
 import { developer, governesses, products } from '../../stories/components/footer/iconsData';
-import styled from '@emotion/styled';
-import { useCookies } from 'react-cookie';
-import useLocalStorage from '../hooks/useLocalStorage';
+import Header from '../../stories/components/header/header';
+import { itemsWebSiteLinks } from '../../stories/components/header/select-link-website/menu-items';
+import useThemeMode from '../hooks/useThemeMode';
+import MainWrapper from './MainWrapper';
+import { ThemeType } from '../enums/theme.enum';
 
-const DARK_SCHEME_QUERY = '(prefers-color-scheme: dark)';
+export type ThemeMode = ThemeType;
+
+const toggleThemeValues = {
+  light: ThemeType.DARK,
+  dark: ThemeType.LIGHT,
+} as Record<string, ThemeType>;
 
-export type ThemeMode = 'light' | 'dark';
 interface ThemeContextType {
   themeMode: ThemeMode;
   toggleTheme: () => void;
+  isLight: boolean;
 }
 const ThemeContext = createContext<ThemeContextType>({} as ThemeContextType);
 const useThemeContext = () => useContext(ThemeContext);
 
 const ThemeProvider = ({ children }: { children: ReactNode }) => {
-  const [cookies] = useCookies(['darkMode']);
-  const userSystemThemePreferenceDark = useMediaQuery(DARK_SCHEME_QUERY);
-
-  const defaultInitialTheme = userSystemThemePreferenceDark && cookies.darkMode === 'true' ? 'dark' : 'light';
+  const { currentTheme: themeMode, handleThemeMode, isLight } = useThemeMode();
 
-  const { state, handleStorageChange: setThemeMode } = useLocalStorage('themeMode', defaultInitialTheme);
-  const themeMode = state as ThemeMode;
   const toggleTheme = () => {
-    switch (themeMode) {
-      case 'light':
-        setThemeMode('dark');
-        break;
-      case 'dark':
-        setThemeMode('light');
-        break;
-      default:
-    }
+    if (themeMode) handleThemeMode(toggleThemeValues[themeMode]);
   };
 
+  const theme = useMemo(() => (isLight ? lightTheme : darkTheme), [themeMode]);
+
+  useEffect(() => {
+    if (themeMode !== undefined) {
+      handleThemeMode(toggleThemeValues[themeMode]);
+    }
+  }, []);
   return (
     <ThemeContext.Provider
       value={{
-        themeMode,
+        themeMode: themeMode as unknown as ThemeMode,
         toggleTheme,
+        isLight: themeMode === ThemeType.LIGHT,
```

Comment:
`isLight` is getting from `useThemeMode` so it can be used here

Reply:
you are right